core/arch/mangle_shared.c

```diff
@@ -1824,7 +1824,7 @@ find_syscall_num(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr)
 
     if (prev == NULL) {
 #if defined(WINDOWS) && defined(X64)
-        if (get_os_version() >= WINDOWS_VERSION_10_1511) {
+        if (get_os_version() >= WINDOWS_VERSION_10_1511 && !DYNAMO_OPTION(inject_x64)) {
```

Comment:
I don't understand: this same code will be there when in 64-bit mode: why disable it?

Reply:
This code messes up with syscall handling, prompting an error on some syscalls when running WoW64 child injected from x64 parent. Disabling this part prevented the wrong deduction of syscall I guess. I did not dive into debugging to understand why this part messes up with syscall on x64->WoW64, but it sure does.

Reply:
It seems like it needs to check the current mode being 64-bit, since the existing behavior will be wanted for 64-bit mode: so it would depend on the mode rather than the -inject_x64 option.